packages/gatsby-source-nacelle/gatsby-node.js

```diff
@@ -32,28 +32,35 @@ exports.sourceNodes = async (gatsbyApi, pluginOptions) => {
     nacelleGraphqlToken,
     nacelleEndpoint,
     contentfulPreviewSpaceId,
-    contentfulPreviewApiToken
+    contentfulPreviewApiToken,
+    nacelleClient
   } = pluginOptions;
 
-  const client = nacelleClient({
-    previewMode: cmsPreviewEnabled(pluginOptions),
-    nacelleSpaceId,
-    nacelleGraphqlToken,
-    nacelleEndpoint,
-    contentfulPreviewSpaceId,
-    contentfulPreviewApiToken
-  });
+  const client = nacelleClient
+    ? nacelleClient
+    : createNacelleClient({
+        previewMode: cmsPreviewEnabled(pluginOptions),
+        nacelleSpaceId,
+        nacelleGraphqlToken,
+        nacelleEndpoint,
+        contentfulPreviewSpaceId,
+        contentfulPreviewApiToken
+      });
```

Comment:
Just a stylistic thing, but would recommend we prefer avoiding an [unneeded ternary](https://eslint.org/docs/rules/no-unneeded-ternary):

```suggestion
  const client = nacelleClient || createNacelleClient({
        previewMode: cmsPreviewEnabled(pluginOptions),
        nacelleSpaceId,
        nacelleGraphqlToken,
        nacelleEndpoint,
        contentfulPreviewSpaceId,
        contentfulPreviewApiToken
      });
```

Reply:
I have mixed feelings on this. In some ways I think the ternary makes intentions a bit clearer than `||`, but nullish coalescing would be even better (but I think it's technically a breaking change because requires node 14+). I've made this change though.